File: src/main/java/net/somrpg/swordofmagic7/TaskUtils.java
package net.somrpg.swordofmagic7;

import org.bukkit.scheduler.BukkitRunnable;

import java.util.concurrent.atomic.AtomicLong;
import java.util.function.LongFunction;

public class TaskUtils {

    private TaskUtils() {
        throw new UnsupportedOperationException("これはユーティリティクラスです。インスタンス化できません");
    }

    public static void runTaskTimerAsync(LongFunction<Boolean> action, long delay, long period) {
        AtomicLong count = new AtomicLong(0);

        new BukkitRunnable() {
            @Override
            public void run() {
                if (Boolean.FALSE.equals(action.apply(count.getAndIncrement()))) {
                    cancel();
                }
            }
        }.runTaskTimerAsynchronously(SomCore.instance, delay, period);
    }
}

File: src/main/kotlin/net/somrpg/swordofmagic7/TaskUtils.kt
package net.somrpg.swordofmagic7

import org.bukkit.scheduler.BukkitRunnable
import java.util.concurrent.atomic.AtomicLong
import java.util.function.LongFunction

object TaskUtils {

    init {
        throw UnsupportedOperationException("これはユーティリティクラスです。インスタンス化できません")
    }

    fun runTaskTimerAsync(action: LongFunction<Boolean>, delay: Long, period: Long) {
        val count = AtomicLong(0)

        object : BukkitRunnable() {
            override fun run() {
                if (action.apply(count.getAndIncrement()) == false) {
                    cancel()
                }
            }
        }.runTaskTimerAsynchronously(SomCore.instance, delay, period)
    }
}
